major_event_log/feeds.py

```diff
@@ -2,101 +2,157 @@
 
 from django.contrib.syndication.views import Feed
 from django.core.urlresolvers import reverse, reverse_lazy
-from django.utils.feedgenerator import Atom1Feed
+from django.utils.feedgenerator import Atom1Feed, rfc3339_date
 from django.core.paginator import Paginator
 
 from .models import Event
 
 
-class MajorEventLogFeed(Atom1Feed):
-    """Custom Atom Feed for Event objects.
+class PaginatedAtom1FeedMixin(object):
+    """Mixin to enable pagination for feeds using Atom1Feed"""
 
-    The only changes between this custom feed and the base Atom Feed
-    are that this feed specifies its mime_type as 'application/xml',
-    pagination is added, the 'alternate' link is removed.
-    """
+    def _create_link_attr(self, rel, page):
+        """Helper function to compose the attributes dictionary argument
+        that handler.addQuickElement accepts.
+        """
+        href = u'{0}?{1}={2}'.format(
+            self.feed['link'], self.feed['page_field'], page)
 
-    mime_type = 'application/xml'
+        return {u'rel': rel, u'href': href}
 
     def add_root_elements(self, handler):
-        handler.addQuickElement(u'title', self.feed['title'])
+        """This method overrides the Atom1Feed.add_root_elements
+        in order to place the additional link elements into the
+        desired location in the document.
+        """
+        handler.addQuickElement('title', self.feed['title'])
         handler.addQuickElement(
-            u'link', '', {u'rel': u'self', u'href': self.feed['feed_url']}
-        )
-        handler.addQuickElement(u'id', self.feed['id'])
-        handler.startElement(u'author', {})
-        handler.addQuickElement(u'name', self.feed['author_name'])
-        handler.addQuickElement(u'uri', self.feed['author_link'])
-        handler.endElement(u'author')
-        handler.addQuickElement(u'subtitle', self.feed['subtitle'])
+            'link', '', {'rel': 'alternate', 'href': self.feed['link']})
+
+        if self.feed['feed_url'] is not None:
+            handler.addQuickElement(
+                'link', '', {'rel': 'self', 'href': self.feed['feed_url']})
```

Comment:
Think this one should also have the closing ')' on the next line. Also, the same thing for the call below this one.